src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -553,7 +553,7 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, ControlValue
                 && !optionValues.some(v => this.matSelect.compareWith(v, previousValue))) {
                 // if a value that was selected before is deselected and not found in the options, it was deselected
                 // due to the filtering, so we restore it.
-                values.push(previousValue);
+                values = [...values, previousValue];
```

Comment:
please do a `values = [...values]` copy before the forEach loop, and then keep the values.push here